release: v1.55.0 (#6023)
apollographql/router: .changesets/fix_bryn_datadog_sample_propagation.md

> ### Fix datadog sample propagation ([PR #6005](https://github.com/apollographql/router/pull/6005))
> 
> #5788 introduced a regression where samping was being set on propagated headers regardless of the sampling decision in the router or upstream.
> 
> This PR reverts the code in question and adds a test to check that a non-sampled request will not result in sampling in the downstream subgraph service.

Reviewer: ```suggestion
### Fix Datadog sample propagation ([PR #6005](https://github.com/apollographql/router/pull/6005))

[PR #5788](https://github.com/apollographql/router/pull/5788) introduced a regression where sampling was set on propagated headers regardless of the sampling decision in the router or upstream.

This PR reverts the code in question and adds a test to check that a non-sampled request doesn't result in sampling in the downstream subgraph service.
```
